### PR TITLE
add search result

### PR DESCRIPTION
**Pull request by yi-fang-yuan**
# Summary

Now each time someone types in the search bar, if you open the side bar. It will displays all the relevant murals that are stored in our database.

Screen Shot 2021-01-26 at 8 48 48 PM

I think what we can do to make the images dynamic is to set a field in the database called imageUrl which stores the url of the respective murals. So in frontend we could just use retrieve that url for each mural or Collection. 

## Related Issues

Issue #52

### CODE REVIEW

**Comment by tedspare** [comment]
Vercel is hanging up on this error:

> src/SideBarSearch/searchCard.tsx Line 3:8:
> `CardMedia` is defined but never used  `@typescript-eslint/no-unused-vars`

Should be a one-line fix then it'll deploy

File: frontend/src/App/App.tsx [hunk 1]
```diff
@@ -63,19 +68,26 @@ function App() {
           signInClick={handleSignin}
           cancelClick={handleCancelSignin}
           open={signingIn} />
-        <Search />
+        <Search searchCallBack={handleSearch}/>
```
Comment (tedspare) [inline frontend/src/App/App.tsx:71]: nice extension of the search functionality

File: frontend/src/SideBarSearch/searchCard.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+const useStyles = makeStyles((theme: Theme) =>
+    createStyles({
+        flexContainer: {
+            display: "flex",
+            flexDirection: "column",
+            alignItems: "flexStart",
+        },
+        card: {
+            width: "450px",
```
Comment (tedspare) [inline frontend/src/SideBarSearch/searchCard.tsx:15]: width should be proportional (%) for various screen sizes

File: frontend/src/SideBarSearch/searchCard.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        },
+        properties: {
+            marginLeft: '5%',
+            display: 'flex',
+            flexDirection: 'column',
+            justifyContent: 'spaceBetween',
+            width: '30%',
+        },
+        type: {
+            marginLeft: '15%'
```
Comment (tedspare) [inline frontend/src/SideBarSearch/searchCard.tsx:31]: use `theme.spacing(1)`, `theme.spacing(2)` etc for consistency

File: frontend/src/SideBarSearch/searchCard.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+                            mural
+                        </Typography>
+                    </Card>)
+
+                })
+            }
+        </div>
+    );
+}
+
```
Comment (tedspare) [inline frontend/src/SideBarSearch/searchCard.tsx:72]: add the ESLint extension to VSCode and set `formatOnSave` to `true` in workspace settings (cmd+,) to clean up whitespace
Screen Shot 2021-01-26 at 9 28 51 PM

**Review by tedspare — APPROVED** [review]
Added some styling comments otherwise LGTM

File: frontend/src/SideBarSearch/searchCard.tsx (hunk 2 above)
Comment (yi-fang-yuan) [reply frontend/src/SideBarSearch/searchCard.tsx:15]: the thing is if I were to put a percentage the sidebar would take the whole screen. 
Screen Shot 2021-01-26 at 9 35 16 PM

File: frontend/src/App/App.tsx [hunk 5]
```diff
@@ -27,6 +28,10 @@ function App() {
     await FirebaseAuth.signOut()
   }
 
+  const handleSearch = (results: any) => {
+    setSearchResult(results);
+  }
```
Comment (JDziewonski98) [inline frontend/src/App/App.tsx:33]: for one line callbacks like this, consider writing them 'in-line' in the component definition like '<Search searchCallBack={(results : any) => setSearchResults(results)}/>' ? IMO is a little cleaner, but just my opinion.

**Pull request merged**